pennylane/resources/first_quantization.py

```diff
@@ -0,0 +1,232 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the functions needed for estimating the number of logical qubits and
+non-Clifford gates for quantum algorithms in first quantization using a plane-wave basis.
+"""
+# pylint: disable= too-many-arguments
+from pennylane import numpy as np
+
+
+def _cost_qrom(lz):
+    r"""Return the minimum number of Toffoli gates needed for erasing the output of a QROM.
+    ￼
+    The expression for computing the cost is taken from
+    [`arXiv:2105.12767 <https://arxiv.org/abs/2105.12767>`_].
+
+    Args:
+        lz (int): sum of the atomic numbers of nuclei
+
+    Returns:
+        int: the minimum cost of erasing the output of a QROM
+
+    **Example**
+
+    >>> _cost_qrom_min(100)
+    21
+    """
+    k_f = np.floor(np.log2(lz) / 2)
+    k_c = np.ceil(np.log2(lz) / 2)
+
+    cost_f = int(2**k_f + np.ceil(2 ** (-1 * k_f) * lz))
+    cost_c = int(2**k_c + np.ceil(2 ** (-1 * k_c) * lz))
+
+    return min(cost_f, cost_c)
+
+
+def unitary_cost(n, eta, omega, error, lamb, br=7, charge=0):
```

Comment:
In this and all following cost functions, I think it will be better for the users if they don't have to pass lambda explicitly, since this should be calculated automatically given the main input parameters `n, eta, omega, error`

Reply:
Will change it when the norm function PR is merged. The cost functions need to get lambda as an input in this PR to pass the tests.